packages/flame/lib/src/sprite_batch.dart

```diff
@@ -216,21 +261,26 @@ class SpriteBatch {
     RSTransform? transform,
     bool flip = false,
     Color? color,
-  }) {
+  }) async {
     final batchItem = BatchItem(
       source: source,
       transform: transform ??= defaultTransform ?? RSTransform(1, 0, 0, 0),
       flip: flip,
       color: color ?? defaultColor,
     );
 
+    if (flip && useAtlas && !_hasFlips) {
+      _hasFlips = true;
+      atlas = await _flippableAtlas;
```

Comment:
I wouldn't use `await` here (nor declare this method `async`) -- this has potential of introducing some subtle bugs like if a user adds several transforms
```dart
batch.addTransform(...);
batch.addTransform(..., flip: true);
batch.addTransform(...);
```
then the second transform will be added in the end, after the 3rd, because it waits for the new atlas to be generated. This would be problematic for users who want their batches to be in the exact order they specified.